GoogleSignIn/Sources/GIDAuthentication.m

```diff
@@ -259,11 +281,23 @@ - (void)doWithFreshTokens:(GIDAuthenticationAction)action {
         });
       }
     }];
+#else // TARGET_OS_OSX
+    NSArray *authenticationHandlerQueue;
+    @synchronized(self->_authenticationHandlerQueue) {
```

Comment:
@petea Peter can you answer this question?

Reply:
The block will capture self instead of the iVar itself. The warning is just to make sure the the developer understand this behaviour. When we add self here the warning will be turn off.

Reply:
To build on what @petea mentioned, it is also the case that `_authenticationHandlerQueue` is declared as an ivar in the implementation, which means the dot syntax won't work here.